src/extension.ts

```diff
@@ -1,45 +1,92 @@
-import {Dash} from './dash';
-import {platform} from 'os';
+import { window, workspace, commands, ExtensionContext, TextEditor, Selection, InputBoxOptions } from 'vscode';
+import { exec } from 'child_process';
+import { Dash } from './dash';
+import { platform } from 'os';
 
 const OS: string = platform();
 
 export function activate(context: ExtensionContext) {
 
-	context.subscriptions.push(commands.registerCommand('extension.dash.specific', () => {
-        searchSpecific();
-	}));
+  context.subscriptions.push(commands.registerCommand('extension.dash.selection', () => {
+    searchSpecific();
+  }));
 
-    context.subscriptions.push(commands.registerCommand('extension.dash.all', () => {
-        searchAll();
-	}));
+  context.subscriptions.push(commands.registerCommand('extension.dash.all', () => {
+    searchAll();
+  }));
+
+  context.subscriptions.push(commands.registerCommand('extension.dash.syntax', () => {
+    searchSyntax();
+  }));
+
+  context.subscriptions.push(commands.registerCommand('extension.dash.customWithSyntax', () => {
+    customWithSyntax();
+  }));
 }
 
 /**
- * Search in dash for specific syntax documentation
+ * Search in dash for selection syntax documentation
  */
 function searchSpecific() {
-    var editor = getEditor();
-    var query = getSelectedText(editor);
-    var languageId = editor.document.languageId;
-    var docsets = getDocsets(languageId);
+  var editor = getEditor();
+  var query = getSelectedText(editor);
+  var languageId = editor.document.languageId;
+  var docsets = getDocsets(languageId);
 
-    var dash = new Dash(OS);
+  var dash = new Dash(OS);
 
-    exec(dash.getCommand(query, docsets));
+  exec(dash.getCommand(query, docsets));
 }
 
 /**
  * Search in dash for all documentation
  */
 function searchAll() {
-    var editor = getEditor();
-    var query = getSelectedText(editor);
+  var editor = getEditor();
+  var query = getSelectedText(editor);
 
-    var dash = new Dash(OS);
+  var dash = new Dash(OS);
+
+  exec(dash.getCommand(query));
+}
+
+/**
+ * Search in dash for editor syntax documentation
+ */
+function searchSyntax() {
+
+  var editor = getEditor();
+  var query = "";
+  var languageId = editor.document.languageId;
+  var docsets = getDocsets(languageId);
+  var dash = new Dash(OS);
+
+  exec(dash.getCommand(query, docsets));
+}
+
+/**
+ * Search in dash for editor syntax documentation with a custom query
+ */
+function customWithSyntax() {
+
+  var editor = getEditor();
+  var languageId = editor.document.languageId;
+  var docsets = getDocsets(languageId);
+  var dash = new Dash(OS);
+
+  var inputOptions: InputBoxOptions = {
+    placeHolder: "Something to search in dash.",
+    prompt: "Enter something to search for in dash."
+  }
+
+  window.showInputBox(inputOptions)
+    .then(function (query: string) {
+      if (query) //If they actually input code
+        exec(dash.getCommand(query, docsets)); //Open it in dash
+    }, function (noQuery: any) {

```

Comment:
After some testing it appears we can. The callback must be a ?: operator so if it isn't there it isn't called, instead of normal javascript where it would shove both params into one variable.